### PR TITLE
E2E: Volume IO Tests CAS-505 CAS-506

### PR DESCRIPTION
**Pull request by blaisedias**
Basic volume IO test iSCSI
Basic volume IO test NVMe-oF TCP

### CODE REVIEW

File: mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	. "github.com/onsi/gomega"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var defTimeoutSecs = "90s"
+
+func TestBasicVolumeIO(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Basic volume IO tests, nvm-of tcp and iSCSI")
```
Comment (jonathan-teh) [inline mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go:19]: Typo: NVMe, capitalisation should be consistent with message on line 53.

File: mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+func basicVolumeIOTest(scName string ) {
```
Comment (jonathan-teh) [inline mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go:22]: Extra space before close bracket (hmm, we seem to be missing a golang linter).

File: mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// JIRA: CAS-505
+// JIRA: CAS-506
```
Comment (jonathan-teh) [inline mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go:2]: Drop these comments.

**Review by jonathan-teh — APPROVED** [review]

File: mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go (hunk 1 above)
Comment (blaisedias) [reply mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go]: Done

File: mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go (hunk 3 above)
Comment (blaisedias) [reply mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go:2]: IIRC these are required so that we know which JIRAs the test is for.

Comment (jonathan-teh) [reply mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go:2]: OK, I see 2 other tests have them too (the first 2 tests I initially looked at didn't).

File: mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+	// Run the fio test
+	common.RunFio(fioPodName, 20)
+
+	// Delete the fio pod
+	err = common.DeletePod(fioPodName)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Delete the volume
+	common.RmPVC(volName, scName)
```
Comment (jkryl) [inline mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go:49]: If anything in the test fails, the PVC will be leaked. Can we clean up PVCs in after-hook instead? And IMHO we should wait until PV, PVC, MSV really disappears before continuing (could be a utility function in common module).

**Review by jkryl — COMMENTED** [review]
If the concern with removing PVCs is addressed then I think there is nothing that prevents us from running this test in ci/cd pipeline. I think that should be done on behalf of this PR.

A short README file would be nice. Just mentioning that all the test does is running fio on nvmf- and iscsi-mayastor-volume. And any other info that might be useful for anyone who wants to run the test without reading the source code can also go there.

**Pull request closed without merging**